src/magic_folder/test/strategies.py

```diff
@@ -439,3 +442,14 @@ def local_snapshots():
         parents_remote=lists(tahoe_lafs_immutable_dir_capabilities()),
         identifier=uuids(),
     )
+
+def path_states():
+    """
+    Build ``PathState`` instances.
+    """
+    return builds(
+        PathState,
+        mtime_ns=integers(min_value=0, max_value=2 ** 31 - 1),
```

Comment:
`mtime_ns` can probably be larger than 2 ** 31 - 1, can't it?  By a factor of 1,000,000,000?

Reply:
What's the motivation to store nanoseconds (instead of e.g. seconds?)

Reply:
The idea is to notice when the file has changed, by storing its metadata. To detect as many changes as possible, we shouldn't throw away any data provided by the OS. I picked ns based on what old magic folders did, but revisiting this, it looks like unix tends to have ns resolution, and windows has 100 ns resolution, so that choice seems reasonable.

Storing less accuracy than is available to us means that there are changes that we won't notice, because the metadata is the same after our rounding.